modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CLibcurlClientCodegen.java

```diff
@@ -908,6 +913,145 @@ public void postProcessFile(File file, String fileType) {
         }
     }
 
+    @Override
+    public ExtendedCodegenOperation fromOperation(String path, String httpMethod, Operation operation, List<Server> servers) {
+        CodegenOperation superOp = super.fromOperation(path, httpMethod, operation, servers);
+        ExtendedCodegenOperation op = new ExtendedCodegenOperation(superOp);
+
+        ApiResponse methodResponse = findMethodResponse(operation.getResponses());
+        if (methodResponse != null) {
+            Map<String, Schema> schemas = ModelUtils.getSchemas(this.openAPI);
+            Schema schema = null;
+            if (schemas != null) {
+                schema = schemas.get(op.returnBaseType);
+            }
+
+            Schema responseSchema = unaliasSchema(ModelUtils.getSchemaFromResponse(methodResponse));
+            if (responseSchema != null) {
+                CodegenProperty cp = fromProperty("response", responseSchema);
+                if (cp != null) {
+                    op.returnTypeIsEnum = cp.isEnum;
+                    op.returnTypeIsBinary = cp.isBinary;
+                    op.returnTypeIsString = cp.isString;
+                    op.returnTypeIsInteger = cp.isInteger;
+                }
+            }
+        }
+
+        return op;
+    }
+
+    class ExtendedCodegenOperation extends CodegenOperation {
```

Comment:
what about using vendor extension instead ?

Reply:
returnTypeIs* seem to be implemented in several other generators, like the typescript one for example. I think I just followed what those generators were doing. I wasn't aware of vendor extensions, but it seems more complicated, since we would have to specify the return type twice for each operation, right?

Reply:
what about using `returnProperty` instead?

```
      "hasDefaultResponse" : false,
      "hasErrorResponseObject" : false,
      "returnProperty" : {
        "openApiType" : "string",
        "baseName" : "response",
        "getter" : "getResponse",
        "setter" : "setResponse",
        "dataType" : "String",
        "datatypeWithEnum" : "String",
        "name" : "response",
        "defaultValueWithParam" : " = data.response;",
        "baseType" : "String",
        "jsonSchema" : "{\n  \"type\" : \"string\"\n}",
        "exclusiveMinimum" : false,
        "exclusiveMaximum" : false,
        "required" : false,
        "deprecated" : false,
        "hasMoreNonReadOnly" : false,
        "isPrimitiveType" : true,
        "isModel" : false,
        "isContainer" : false,
        "isString" : true,
        "isNumeric" : false,
        "isInteger" : false,
        "isShort" : false,
        "isLong" : false,
        "isUnboundedInteger" : false,
        "isNumber" : false,
        "isFloat" : false,
        "isDouble" : false,
        "isDecimal" : false,
        "isByteArray" : false,

```

obtained via debugOperations, e.g.
```
./bin/generate-samples.sh ./bin/configs/java-native-echo-api.yaml -- --global-property debugOperations=true
```

Reply:
That makes sense. I used `returnTypeIs*` because it's what typescript uses for arrays and models, and it also matches `returnTypeIsPrimitive`, but if you prefer I'll rework this to use `returnProperty` instead.

Reply:
This is getting a little messy. The main problem is that I need the `{{{.}}}` that gives the type name to remain outside `returnProperty`, so templates that now look like
```
{{^returnTypeIsEnum}}{{{.}}}_t*{{/returnTypeIsEnum}}{{#returnTypeIsEnum}}{{projectName}}_{{{.}}}_{{returnEnumName}}_e{{/returnTypeIsEnum}}
```

will become

```
{{#returnProperty}}{{#isEnum}}{{projectName}}_{{/isEnum}}{{/returnProperty}}{{{.}}}{{#returnProperty}}{{^isEnum}}_t*{{/isEnum}}{{#isEnum}}_{{returnEnumName}}_e{{/isEnum}}{{/returnProperty}}
```

I think I prefer my original version, but if you disagree please let me know and I'll convert everything to the second version and resubmit the pull request.

Reply:
> The main problem is that I need the {{{.}}} that gives the type name

You mean `{{{dataType}}}`?

Reply:
Sorry, I thought it was clear from the contest. I mean the `{{{.}}}` inside a `{{#returnType}}` block (like the one near the top of `api-header.mustache`, those two code blocks in my post were from that block).

Reply:
What about using `{{{returnType}}}` directly instead of `{{{.}}}` inside `{{#returnType}} ... {{/returnType}}` ?

Reply:
Thanks, that was very helpful. I just submitted a new version with your suggested changes here: https://github.com/OpenAPITools/openapi-generator/pull/14434

Reply:
@eafer thanks for taking my suggestion.

We want to move away from `returnTypeIs*` and use `returnProperty` instead as `returnProperty` should pretty much cover all cases.